run.py

```diff
@@ -37,55 +39,71 @@ def load_column_files(config, sql_sync):
 
     cur.close()
-    conn.close()
-    sql_sync.create_joined_rows()
+    raw_conn.close()
+    sql_sync.create_joined_rows(nonbib_db_conn)
 
 
+def row2dict(row):
+    """ from https://stackoverflow.com/questions/1958219/convert-sqlalchemy-row-object-to-python-dict"""
+    d = {}
+    for column in row.__table__.columns:
+        d[column.name] = getattr(row, column.name)
+    return d
 
-def nonbib_to_master_pipeline(schema, batch_size=1):
+
+def nonbib_to_master_pipeline(nonbib_engine, schema, batch_size=1):
     """send all nonbib data to queue for delivery to master pipeline"""
-    nonbib = row_view.SqlSync(schema)
-    connection = nonbib.engine.connect()
-    s = select([nonbib.table])
-    results = connection.execute(s)
+    global config
+    Session = sessionmaker(bind=nonbib_engine)
+    session = Session()
+    session.execute('set search_path to {}'.format(schema))
     tmp = []
-    for current_row in results:
-        current_row = dict(current_row)
+    i = 0
+    max_rows = config['MAX_ROWS']
+    for current_row in session.query(models.NonBibTable).yield_per(100):
+        current_row = row2dict(current_row)
         rec = NonBibRecord(**current_row)
         tmp.append(rec._data)
+        i += 1
+        if max_rows > 0 and i >= max_rows:
+            break
         if len(tmp) >= batch_size:
             recs = NonBibRecordList()
             recs.nonbib_records.extend(tmp)
             tmp = []
-            logger.debug("Calling 'app.forward_message' with '%s' items", len(recs.nonbib_records))
+            logger.info("Calling 'app.forward_message' count = '%s'", i)
             task_output_results.delay(recs)
 
     if len(tmp) > 0:
         recs = NonBibRecordList()
         recs.nonbib_records.extend(tmp)
-        logger.debug("Calling 'app.forward_message' with final '%s' items", len(recs.nonbib_records))
+        logger.info("Calling 'app.forward_message' with count = '%s'", i)
         task_output_results.delay(recs)
+    session.close()
 
 
-
-def nonbib_delta_to_master_pipeline(schema, batch_size=1):
+def nonbib_delta_to_master_pipeline(nonbib_engine, schema, batch_size=1):
     """send data for changed bibcodes to master pipeline
 
     the delta table was computed by comparing to sets of nonbib data
     perhaps ingested on succesive days"""
-    nonbib = row_view.SqlSync(schema)
-    connection = nonbib.engine.connect()
-    delta_table = nonbib.get_delta_table()
-    s = select([delta_table])
-    results = connection.execute(s)
+    global config
+    Session = sessionmaker(bind=nonbib_engine)
+    session = Session()
+    session.execute('set search_path to {}'.format(schema))
     tmp = []
-    for current_delta in results:
-        row = nonbib.get_row_view(current_delta['bibcode'])
-        rec = NonBibRecord(**dict(row))
+    i = 0
+    max_rows = config['MAX_ROWS']
+    for current_delta in session.query(models.DeltaTable).yield_per(100):
+        row = nonbib.get_row_view(current_delta.bibcode)
```

Comment:
tihs code is missing, also - can you try to find a better name for it? it is not clear what it's supposed to do (and on what)

i'm also seeing it used as a classmethod and as in this instance as a function (of a module); there is some confusion

Reply:
get_row_view renamed to get_by_bibcode
https://github.com/spacemansteve/AdsDataSqlSync/blob/master/adsdata/nonbib.py#L197